ddtrace/utils/cache.py

```diff
@@ -0,0 +1,85 @@
+from threading import RLock
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
+
+
+miss = object()
+
+T = TypeVar("T")
+S = TypeVar("S")
+F = Callable[[T], S]
+M = Callable[[Any, T], S]
+
+
+def cached(maxsize=256):
+    # type: (int) -> Callable[[F], F]
+    """
+    Decorator for caching the result of functions with a single argument.
+
+    The strategy is MFU, meaning that only the most frequently used values are
+    retained. The amortized cost of shrinking the cache when it grows behond
```

Comment:
```suggestion
    retained. The amortized cost of shrinking the cache when it grows beyond
```

Reply:
Thanks! I think I need to invest on a vocabulary 😁